docs/sequence-diagrams/sidecar.puml

```diff
@@ -0,0 +1,8 @@
+@startuml
+thanos_sidecar -> prometheus_data_dir: On init create thanos.shipper.json 
```

Comment:
This roughly sense, just maybe `cloud_storage` -> `object storage`

Also `Write block to storage` cannot be just `data dir` to itself, I would introduce another actor: Prometheus (: 

and there are many more interactions like getting flags or external labels. But this is good a start. I would say splitting PR and focusing on each component would be the easiest way (: